src/common.rs

```diff
@@ -1,21 +1,30 @@
 //! Contains common utility & convenience functions.
 //! All functions here are intended for internal use only.
-use crate::errors::Result;
+use crate::errors::Result as CognativeResult;
```

Comment:
there is a typo in this name (Cogn**a**tiveResult) but anyway I don't like this much. Can we just stick with **Result**?